sub/sd_ass.c

```diff
@@ -259,8 +261,12 @@ static void decode(struct sd *sd, struct demux_packet *packet)
             packet->duration = UNKNOWN_DURATION;
         }
         char **r = lavc_conv_decode(ctx->converter, packet);
-        for (int n = 0; r && r[n]; n++)
-            ass_process_data(track, r[n], strlen(r[n]));
+        for (int n = 0; r && r[n]; n++) {
+            if (sd->opts->sub_filter_SDH)
+                filter_SDH(sd, track->event_format, 0, r[n] , sd->opts->sub_filter_SDH_harder);
```

Comment:
This permutes the buffer allocated/provided by libavcodec. I'm not sure if that's allowed, but it sure isn't nice.